.github/workflows/testflight.yaml

```diff
@@ -0,0 +1,80 @@
+name: Build & Deploy iOS TestFlight
+
+on:
+  create:
+    tags:
+defaults:
+  run:
+    shell: bash
+jobs:
+  build:
+    runs-on: macos-latest
+    env:
+      working-directory-ios: ./ios
+    steps:
+    - uses: actions/checkout@v2
+    - uses: maxim-lobanov/setup-xcode@v1
```

Comment:
Given all these secrets in this workflow it's probably best to checkout a specific sha of each of these actions. So they are not free to change those tags from underneath us and potentially cause problems (like add malicious code). This assume the actions have been vetted already.

`DISTRIBUTION_CERTIFICATE_PASSWORD` is not the same as the production account, is it?

Not sure having this on a public, shared runner is a good idea. Maybe @mitsuhiko can comment.

Reply:
Yes please. No unpinned github actions usage!

Reply:
@bruno-garcia This is the password that you specify when exporting the Distribution Certificate from your local Keychain. The IPA is then signed with this certificate so we can distribute it to the AppStore or TestFlight.

Since this is all in Secrets, only collaborators can run this workflow. So we should be good as long as no secret is leaked. Should this be the case at some point, then we can at least revoke the Distribution Profile.

Same with the secrets used to generate the API Key for the AppStore Connect API which we use to increment the build number and upload to TestFlight.

Reply:
The owner of the repo `maxim-lobanov/setup-xcode@v1` for example can get access to these keys if they later decide to change what `v1` points to and add malicious code that copies our secrets.

This is only 1 possible attack vector that can be done here. Also is that anyone with write access on the repo who not necessarily should have rights to publish the app can also change the workflow yaml and leak the secret.

The first problem is often dealt with by dorking the action or pinning to a commit sha. The second is by having a dedicated repo just for releasing things (sentry is doing that for SDKs).

Reply:
I have added the commit pinning when finishing up the setup yesterday. The other concerns are a problem though, great feedback. 🙇 What is the best way to proceed here, should we discuss introducing a private repo for releasing iOS/Android as suggested? @bruno-garcia  @HazAT

Reply:
I'll defer this one to @mitsuhiko 

I honestly think it's OK to have it here if we pin the actions. We need to be aware that giving someone write access to the repo means they could get a hold of any secrets. With that in mind, we're good. But it's not my call to make.